### ISSUE TITLE
NrSmallGroups undocumented

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#407]
`NrSmallGroups` is a synonym of `NumberSmallGroups` but entering `?NrSmallGroups` results in this:
```
gap> ?NrSmallGroups
Help: no matching entry found
Help: 'NrSmallGroups' is currently undocumented.
      For details, try ?Undocumented Variables
```
Entering the full name works fine:
```
gap> ?NumberSmallGroups
Help: several entries match this topic - type ?2 to get match [2]

[1] Reference: NumberSmallGroups
[2] SglPPow (not loaded): NumberSmallGroups
```

`NrSmallGroups` is mentioned in many places and should be documented. 

This issue is labeled as "newcomer-friendly" and maybe suitable for someone trying to use GAPDoc for GAP documentation or making their first pull request.

**Comment by fingolfin** [comment]
One could also argue the reverse, namely that we whould get rid of `NrSmallGroups` in as many places as possible, and only keep it as a synonym (marked as obsolete) for backwards compatibility. Note that I say "could" -- what I want to say here, I guess, is that we should justify why we prefer one resolution over another.

It also seems weird to only discuss this particular case, considering that there are multiple similar cases. Just look at functions starting with "Number" vs "Nr". It seems "Nr" is used more frequently in the meaning it has as in `NumberSmallGroups`; but this is IMHO rather "un-GAPish", as we do not use such abbreviations elsewhere.

Another concrete example is `NumberPerfectLibraryGroups` vs. `NrPerfectLibraryGroups`. Only the former is documented. Ironically, the latter occurs in manual entry for `SizesPerfectGroups`

So I think we should either "always" allow both "NrFOO" and "NumberFOO" and document both. Or always push for just one of the two names. It would be confusing to treat this in one way for some methods, and in another for some others.

And then we could go beyond. Just consider
BaseIntMat, ComplementIntMat, ...
vs.
TriangulizeIntegerMat, SmithNormalFormIntegerMat, ...

So sometimes it is IntMat, sometimes IntegerMat. Here, too, we should either always allow both; or standardize on one of the two (and of course keep synonyms for the old names, to ensure backwards compatibility).

Personally, I'd prefer choosing a single name and sticking with that. Offering many alternative names serves no purpose, except allowing us to avoid making a decision...

**Comment by stevelinton** [comment]
What might be quite cool (or might be a daft idea) would be to have some abbreviations recognised by auto-complete, so NrSmall<tab> would expand to NumberSmallGroups.

**Comment by alex-konovalov** [comment]
@fingolfin and of course another example is that we have widely known NrConjugacyClasses but no its  NumberConjugacyClasses counterpart.

@stevelinton I haven't look at auto-complete, but I know how to do the following: any query to help system containing Number will look for same entries with Nr and vice versa. This is used in lib/transatl.g to search for both British and American spellings.

**Comment by frankluebeck** [comment]
The inconsistencies mentioned here are mainly a relic from the GAP3 -> GAP4 change.

A systematic cleanup would be nice. It could be done along these lines:

1. Identify the abbreviated and corresponding parts of words which are used:
  * Nr <-> Number
  *  Int  <-> Integer
  *  Mat <-> Matrix
  *  Perm <-> Permutation
  *  ???
2. Decide which one we prefer. (I guess the full word version.)
3. Make a list of existing function names with abbreviated part.
4. Change all names on the list in 3. in the GAP library and documentation to the full word version.
5. Add a new file to the library, say `lib/deprecated.g`,  and declare all names on the list from 3. as synonyms of the full word versions. 
6. Add a section "Deprecated function names" to the reference manual which lists the names from 3., somehow clearly marked as deprecated and with an index entry, and the names one should use instead.

Of course, 4. and 5. are for backward compatibility of existing code. I would say that they can just stay forever and there is no need to move entries from `deprecated.g` to `obsolete.g`.

**Comment by frankluebeck** [comment]
@stevelinton said:
> What might be quite cool (or might be a daft idea) would be to have some abbreviations recognised by auto-complete, so NrSmall would expand to NumberSmallGroups.

With the current implementation of the completion mechanism this wouldn't be difficult to do.

**Comment by fingolfin** [comment]
The original issue reported here now is perhaps something that should be added to the `smallgrp` package repository.

The more elaborate plans afterwards are different of course, but then perhaps the issue title should be changed?

I am also going to remove the "good first issue label".

**Comment by alex-konovalov** [comment]
Agree. Created https://github.com/gap-packages/smallgrp/issues/25 and changed issue title.
